nakama/main.go

```diff
@@ -115,6 +121,38 @@ func initReceiptDispatcher(log runtime.Logger) error {
 	return nil
 }
 
+func initEventHub(ctx context.Context, log runtime.Logger, nk runtime.NakamaModule) error {
+	globalEventHub, err := createEventHub()
+	if err != nil {
+		return err
+	}
+	go func() {
+		_ = globalEventHub.dispatch(log)
+		if err != nil {
+			log.Error("error initializing eventHub: %s", err.Error())
+		}
+	}()
+
+	//for now send to everybody via notifications.
+	go func() {
+		channel := globalEventHub.subscribe("main")
+		for !globalEventHub.didShutdown.Load() {
```

Comment:
This can be changed to 
```
for event := range channel {
...
}
```
Which also removes the select statement and the didShutdown checks. The dispatch function above closes all subscribed channels when it is shut down.

Reply:
You might have seen this after I made the change. But there's a time out of 2 seconds using the select statement. If a channel blocks, every two seconds the loop will check shutdown, so a blocking channel will not stop a shutdown.

Reply:
no wait nm. I see what you're saying.